src/Bicep.LangServer.IntegrationTests/TelemetryTests.cs

```diff
@@ -623,8 +620,9 @@ private async Task<BicepTelemetryEvent> GetTelemetryEventForBicepConfigChange(st
 
             client.TextDocument.DidOpenTextDocument(TextDocumentParamHelper.CreateDidOpenDocumentParams(bicepConfigUri, prevBicepConfigFileContents, 1));
 
-            var diagsParams = await diagsListener.WaitNext();
-            diagsParams.Uri.Should().Be(documentUri);
+            // There's no good way to wait between document open and save. And looks like omnisharp swallows DidSaveTextDocument(..) event
```

Comment:
I'd rather avoid this if possible, takes a long time and still will be flaky.  Maybe add an event to client.TextDocument on document saved that you can listen to, or inject a semaphor to be triggered?

@majastrz Any good ways you guys have solved this before?

Reply:
The code for what we did is in the `IntegrationTestHelper`. We basically wait until diagnostics get published after file open notification gets sent with a timeout in case the server is busted. Might be possible to do something similar here as well.

Reply:
I looked through this and there isn't an obvious way to do it without some additional refactoring. However, this will bite us eventually and cause test flakiness. Let's fix this after this PR goes in to unblock the release.

My idea is basically the following:
- These tests don't need to be full integration tests. We can manually instantiate enough of the components to interact with the document sync handler to trigger our relevant scenarios.
- The telemetry logic from the sync handler and the various helpers could be moved into a new non-static class that is injected as a dependency of the doc sync handler.
- The remaining telemetry logic in doc sync handler would be super simple just call equivalent methods on the new class.
- I think at that point you'd be able to mock enough of the dependencies to remove the Task.Delay() and still get enough of the validation.